### PR TITLE
refactor(internal): rename non-intuitive `internal/alloy` to more comprehensive `internal/runtime`

### LINKED ISSUE #737: Find a better name for the github.com/grafana/alloy/internal/alloy package
### Request

The github.com/grafana/alloy/internal/alloy package is a bit repetitive, and probably isn't the best name. I'd like to find a better, more descriptive name, for the package. `github.com/grafana/alloy/internal/controller` isn't viable as there's already a package called `controller` (`github.com/grafana/alloy/internal/alloy/internal/controller`). 

@mattdurham had used the term "runtime" in the past to refer to the overall Alloy system (including the controller), so maybe `github.com/grafana/alloy/internal/runtime` is a better name here. 

### Use case

Make packages more descriptive to help developers find/understand code.

### PR DESCRIPTION
**Pull request by hainenber**
#### Which issue(s) this PR fixes

Closes #737

### CODE REVIEW

File: internal/runtime/module.go
Comment (rfratto) [inline internal/runtime/module.go:1]: The package name here is still declared as `package alloy` but it should ideally match the folder name and be `package runtime`

File: internal/alloycli/cmd_run.go [hunk 1]
```diff
@@ -19,16 +19,16 @@ import (
...
-	"github.com/grafana/alloy/internal/alloy/logging"
-	"github.com/grafana/alloy/internal/alloy/logging/level"
-	"github.com/grafana/alloy/internal/alloy/tracing"
 	"github.com/grafana/alloy/internal/alloyseed"
 	"github.com/grafana/alloy/internal/boringcrypto"
 	"github.com/grafana/alloy/internal/component"
 	"github.com/grafana/alloy/internal/converter"
 	convert_diag "github.com/grafana/alloy/internal/converter/diag"
 	"github.com/grafana/alloy/internal/featuregate"
+	alloyRuntime "github.com/grafana/alloy/internal/runtime"
```
Comment (rfratto) [inline internal/alloycli/cmd_run.go:28]: Given my previous comment, if the package name is changed to runtime there won't be a need for the `alloyRuntime` import alias (unless there's already an imported package called runtime)

File: internal/alloycli/cmd_run.go [hunk 2]
```diff
@@ -390,7 +390,7 @@ func (fr *alloyRun) Run(configPath string) error {
 }
 
 // getEnabledComponentsFunc returns a function that gets the current enabled components
-func getEnabledComponentsFunc(f *alloy.Alloy) func() map[string]interface{} {
+func getEnabledComponentsFunc(f *alloyRuntime.Alloy) func() map[string]interface{} {
```
Comment (rfratto) [inline internal/alloycli/cmd_run.go:393]: The struct being named Alloy (`alloyRuntime.Alloy` / `runtime.Alloy`, once the package path and imports are changed) feels a bit weird to me now that the package is named differently. Maybe it makes sense to rename the struct to `Runtime` so it's used as `runtime.Runtime`?

**Review by rfratto — COMMENTED** [review]
Thanks for going through the effort of this refactoring! The changes look reasonable, but I have a few comments before approving.

File: internal/runtime/module.go
Comment (hainenber) [reply internal/runtime/module.go:1]: I've fixed in HEAD

File: internal/alloycli/cmd_run.go (hunk 2 above)
Comment (hainenber) [reply internal/alloycli/cmd_run.go]: Also fixed in HEAD :D

**Review by rfratto — APPROVED** [review]
LGTM, thanks!

**Pull request merged**